cmd/up.go

```diff
@@ -174,20 +220,32 @@ func reconnectPortForward(ctx context.Context, wg *sync.WaitGroup, d *appsv1.Dep
 	}
 
 	if err := pf.Start(ctx, wg, client, restConfig, pod); err != nil {
-		return fmt.Errorf("couldn't start the connection to your cluster: %s", err)
+		return fmt.Errorf("couldn't connect to your cluster: %s", err)
 	}
 
 	log.Infof("reconnected port-forwarder-%d:%d", pf.LocalPort, pf.RemotePort)
 
 	return nil
 }
 
-func shutdown(cancel context.CancelFunc, wg *sync.WaitGroup) {
+// Shutdown runs the cancellation sequence. It will wait for all tasks to finish for up to 500 milliseconds
+func Shutdown(cancel context.CancelFunc, wg *sync.WaitGroup) {
 	log.Debugf("cancelling context")
 	cancel()
 
 	log.Debugf("waiting for tasks for be done")
-	wg.Wait()
-
-	log.Debugf("completed shutdown sequence")
+	done := make(chan struct{})
+	go func() {
+		wg.Wait()
+		close(done)
+	}()
+
+	select {
```

Comment:
there's a race condition in the port forwarding code (I think) that's causing this to block sometimes). I'm going to fix that on a separate PR. Either way, I think we should time out this, there's no reason to block the exit for more than a second or so.